content/blog/2021-11-10-agent.md

```diff
@@ -0,0 +1,156 @@
+
+Note that this example uses Thanos in receive mode as the remote storage. Nowadays, you can use plenty of other projects that are compatible with remote write API.
+
+So if remote writing works fine, why do we add special Agent mode to Prometheus?
+
+## Prometheus Agent Mode
+
+From Prometheus `v2.32.0` (next release), everyone will be able to run Prometheus binary with an experimental `--enable-feature=agent` flag. If you want to try it before the release, feel free to use [Prometheus v2.32.0-beta.0](https://github.com/prometheus/prometheus/releases/tag/v2.32.0-beta.0) or use our `quay.io/prometheus/prometheus:v2.32.0-beta.0` image.
+
+The Agent mode optimizes Prometheus for the remote write use case. It disables querying, alerting and local storage and replaces it with customized TSDB WAL. Everything else stays the same: scraping logic, service discovery and related configuration. It can be used as a drop-in replacement for Prometheus if you want to just forward your data to remote Prometheus and any other Remote Write compliant project. In essence it looks like this:
+
+![Prometheus agent](/assets/blog/2021-11-10/agent.png)
+
+The best part about Prometheus Agent is that it's built into Prometheus. Same scraping APIs, same semantics, same configuration and discovery mechanism.
+
+What benefits of using Agent mode if you plan not to query or alert on data locally and stream metrics outside? There are a few:
+
+First of all, efficiency. Our customized Agent TSDB WAL removes the data immediately after successful writes. If it cannot reach the remote endpoint, it persists the data temporarily on the disk until the remote endpoint is back online. This is currently limited to 2h buffer only, similar to non-agent Prometheus, [hopefully unblocked soon](https://github.com/prometheus/prometheus/issues/9607). This means that we don't need to build chunks of data in memory. We don't need to maintain a full index for querying purposes. Essentially Agent uses a fraction of resources normal Prometheus would use in a similar situation.
+
+Does this efficiency matter? Yes! As we mentioned, every GB of memory and every CPU core used on the edge clusters matters for some deployments. On the other hand, the paradigm of performing monitoring using metrics is quite mature these days. This means that the more relevant metrics with more cardinality you can ship for the same cost - the better.
+
+> NOTE: With the introduction of the Agent Mode, the original Prometheus server mode still stays as the recommended, stable and maintained mode. Agent Mode with remote storage brings additional complexity. Use with care.
```

Comment:
```suggestion
> NOTE: With the introduction of the Agent mode, the original Prometheus server mode still stays as the recommended, stable and maintained mode. Agent Mode with remote storage brings additional complexity. Use with care.
```

Mostly you spell "mode" lower-cased, but sometimes not. Would be good to make it consistent throughout.